main.go

```diff
@@ -147,6 +155,92 @@ func doInitrdSetup(ctx *cli.Context) error {
 	return t.InitrdSetup()
 }
 
+func generateKeyPair(newUUID string) error {
+	privKey, err := rsa.GenerateKey(rand.Reader, 2048)
+	if err != nil {
+		return err
+	}
+
+	err = privKey.Validate()
+	if err != nil {
+		return err
+	}
+
+	CN := fmt.Sprintf("manifest PRODUCT:%s", newUUID)
+	template := x509.CertificateRequest {
+		Subject: pkix.Name {
+			CommonName: CN,
+		},
+	}
+
+	// Create a CSR with the new key
+	newCSR, err := x509.CreateCertificateRequest(rand.Reader, &template, privKey)
+	if err != nil {
+		return err
+	}
+
+	keyPEM := pem.EncodeToMemory (
+		&pem.Block {
+			Type: "RSA PRIVATE KEY",
+			Bytes: x509.MarshalPKCS1PrivateKey(privKey),
+		},
+	)
+
+	csrPEM := pem.EncodeToMemory (
+		&pem.Block {
+			Type: "CERTIFICATE REQUEST",
+			Bytes: newCSR,
+		},
+	)
+
+	//  get trust dir
+	trustDir, err := getTrustPath()
+        if err != nil {
+		return err
+        }
+
+        defer func() {
+                if err != nil {
+                        os.Remove(filepath.Join(trustDir,"manifest.key"))
+                        os.Remove(filepath.Join(trustDir,"manifest.csr"))
+                        os.Remove(filepath.Join(trustDir,"uuid"))
+                }
+        }()
+
+	// Save private key to trust dir
+	err = os.WriteFile(filepath.Join(trustDir, "manifest.key"), keyPEM, 0600)
+	if err != nil {
+		return err
+	}
+
+	// Save CSR to trust dir
+
+	err = os.WriteFile(filepath.Join(trustDir, "manifest.csr"), csrPEM, 0660)
```

Comment:
Sorry, let's make this and the next one 0640.

Reply:
done

Reply:
it still shows as 0660 here.  Have you force-pushed?